source/Halibut/ServiceModel/ServiceInvoker.cs

```diff
@@ -17,26 +18,34 @@ public ServiceInvoker(IServiceFactory factory)
             this.factory = factory;
         }
 
-        public ResponseMessage Invoke(RequestMessage requestMessage)
+        public async Task<ResponseMessage> Invoke(RequestMessage requestMessage)
         {
             using (var lease = factory.CreateService(requestMessage.ServiceName))
             {
                 var methods = lease.Service.GetType().GetMethods().Where(m => string.Equals(m.Name, requestMessage.MethodName, StringComparison.OrdinalIgnoreCase)).ToList();
                 if (methods.Count == 0)
                 {
-                    return ResponseMessage.FromError(requestMessage, string.Format("Service {0}::{1} not found", lease.Service.GetType().FullName, requestMessage.MethodName));
+                    return ResponseMessage.FromError(requestMessage, $"Service {lease.Service.GetType().FullName}::{requestMessage.MethodName} not found");
                 }
 
                 var method = SelectMethod(methods, requestMessage);
                 var args = GetArguments(requestMessage, method);
-                var result = method.Invoke(lease.Service, args);
+                var result = await InvokeAsync(method, lease.Service, args).ConfigureAwait(false);
                 return ResponseMessage.FromResult(requestMessage, result);
             }
         }
 
+        static async Task<object> InvokeAsync(MethodBase methodBase, object obj, params object[] parameters)
+        {
+            var task = (Task)methodBase.Invoke(obj, parameters);
+            await task.ConfigureAwait(false);
+            var resultProperty = task.GetType().GetProperty("Result");
```

Comment:
Is there a more efficient way to do this? Also this assumes it is always a Task<TResult>, is that safe?